confluent-cloud-rest-produce-schema-registry/README.md

```diff
@@ -0,0 +1,187 @@
+
+```shell
+confluent login --prompt
+```
+
+Create a Kafka topic named `orders`:
+
+```shell
+confluent kafka topic create orders
+```
+
+Associate the schema in the file `order.avsc` with the record values in the `orders` topic:
+
+```shell
+confluent schema-registry schema create --subject orders-value --schema order.avsc --type avro
+```
+
+The schema defines an order with a customer ID, product ID, and product name:
+
+```json
+{
+  "type": "record",
+  "namespace": "io.confluent.developer",
+  "name": "Order",
+  "fields": [
+    {
+      "name": "product_id",
+      "type": "int"
+    },
+    {
+      "name": "customer_id",
+      "type": "int"
+    },
+    {
+      "name": "product_name",
+      "type": "string"
+    }
+  ]
+}
+```
+
+## Generate base64-encoded message
+
+Following the Produce API [data payload specification](https://docs.confluent.io/cloud/current/kafka-rest/kafka-rest-cc.html#data-payload-specification), we must send base64-encoded binary data that conforms to this [wire format](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/index.html#wire-format).
+
+The GitHub repo contains a sample Python script to help with this. First, instantiate a byte buffer and write the magic byte (0), followed by the schema ID in the next 4 bytes:
+
+```python
+buffer = io.BytesIO()
+
+# write magic byte (zero)
+buffer.write(b'\x00')
+
+# write schema ID in next 4 bytes
+buffer.write(args.schema_id.to_bytes(4))
+```
+
+Next, write the Avro-formatted data from `payload.json` into the same buffer. The sample payload looks like:
+
+```json
+{
+  "product_id": 19287,
+  "customer_id": 8737,
+  "product_name": "AirSwift Light Sneakers"
+}
+```
+
+To write this binary-encoded data:
+
+```python
+# write the Avro-formatted data
+DatumWriter(schema).write(record, BinaryEncoder(buffer))
+```
+
+Finally, base64-encode the entire payload and print it:
+
+```python
+print(base64.b64encode(buffer.getvalue()).decode("utf-8"))
+```
+
+To run the script and generate the encoded payload:
+
+1. Get the schema ID:
+   ```shell
+   confluent schema-registry schema describe  --subject orders-value --version latest | grep "Schema ID"
+   ```
+2. Install the `avro` package and run the script, passing the schema ID from the previous step:
+   ```shell
+   pip install avro
+   PAYLOAD=$(./generate_sample_avro_value.py --schema-path ./order.avsc \
+                                             --payload-path ./payload.json \
+                                             --schema-id <SCHEMA ID>)
+   ```
+   This prints the payload that we will post to the Produce REST API:
+   ```noformat
+   $ echo $PAYLOAD
+   AAABhqOurQLCiAEuQWlyU3dpZnQgTGlnaHQgU25lYWtlcnM=
+   ```
+
+## Send message to Produce API
+
+With the base64-encoded message ready, set the environment variables `REST_ENDPOINT` and `CLUSTER_ID` by describing the cluster:
+
+```shell
+read -r CLUSTER_ID REST_ENDPOINT < \
```

Comment:
I'm getting this outout here --  confluent-cloud-rest-produce-schema-registry % read -r CLUSTER_ID REST_ENDPOINT < \
    <(confluent kafka cluster describe -o json | jq -cr '"\(.id) \(.rest_endpoint)"')
Error: no Kafka cluster selected

Suggestions:
    You must provide the cluster ID argument or set an active Kafka cluster in your context with confluent kafka cluster use.

Reply:
this should be resolved now too